core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -175,6 +175,29 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
     sendAndVerifyTimestamp(producer, TimestampType.CREATE_TIME)
   }
 
+  protected def sendAndVerify(producer: KafkaProducer[Array[Byte], Array[Byte]], numRecords: Int = numRecords) {
+    val partition = new Integer(0)
+    try {
+      TestUtils.createTopic(zkUtils, topic, 1, 2, servers)
+
+      val recordAndFutures = for (i <- 1 to numRecords) yield {
+        val record = new ProducerRecord(topic, partition, "key".getBytes, "value".getBytes)
+        (record, producer.send(record))
+      }
+      producer.close(20000L, TimeUnit.MILLISECONDS)
```

Comment:
Should we allow this timeout to be passed as a parameter (with a default) as it may need to be different depending on `numRecords`?